### ISSUE TITLE
TypeError: getSnapshot is not a function

### ISSUE THREAD

**Issue by Kripu77** [pmndrs/zustand#963]
Hi, I'm new to the Zustand Space, could you please help me understand what actually I'm doing wrong?

I have been just following up on the code sample provided on the Zustand NPM page. However, I keep on getting the same error. Please help me understand what actually is wrong or what I need to fix.

To be more concise, I created a store and binded my store to my components as per the example is provided in the docs.

**Comment by dai-shi** [comment]
Hi, would you share the full error log? Can you try codesandbox to see if it works?

**Comment by Kripu77** [comment]
HI Dai, 

I have a store component which has the following code which I took as a reference from the docshomepage:

 import create from 'zustand'

const useStore = create(set => ({
  bears: 0,
  increasePopulation: () => set(state => ({ bears: state.bears + 1 })),
  removeAllBears: () => set({ bears: 0 })
}))

Additionally, I have a Display component which has the following: 
import React from 'react'
import { useStore } from 'zustand'

const Home = () => {
 const bears = useStore((state) => state.bears);
 return <h1>{bears} around here ...</h1>;
  
}

export default Home;


Howerver, upon rendering the Home component, I get:

TypeError: getSnapshot is not a function
getSnapshotWithSelector

 var maybeGetServerSnapshot = getServerSnapshot === undefined ? null : getServerSnapshot;
TypeError: getSnapshot is not a function
getSnapshotWithSelector
http://localhost:3000/static/js/vendors~main.chunk.js:82269:35
  82266 | var maybeGetServerSnapshot = getServerSnapshot === undefined ? null : getServerSnapshot;
  82267 | 
  82268 | var getSnapshotWithSelector = function () {
> 82269 |   return memoizedSelector(getSnapshot());
        |                           ^  82270 | };
  82271 | 
  82272 | var getServerSnapshotWithSelector = maybeGetServerSnapshot === null ? undefined : function () {
View source
useSyncExternalStore
http://localhost:3000/static/js/vendors~main.chunk.js:82016:19
  82013 | // implementation details, most importantly that updates are
  82014 | // always synchronous.
  82015 | 
> 82016 | var value = getSnapshot();
        |             ^  82017 | {
  82018 |   if (!didWarnUncachedGetSnapshot) {
  82019 |     var cachedValue = getSnapshot();
View source
useSyncExternalStoreWithSelector
http://localhost:3000/static/js/vendors~main.chunk.js:82280:19
  82277 |     getSelection = _useMemo[0],
  82278 |     getServerSelection = _useMemo[1];
  82279 | 
> 82280 | var value = useSyncExternalStore(subscribe, getSelection, getServerSelection);
        |             ^  82281 | useEffect(function () {
  82282 |   inst.hasValue = true;
  82283 |   inst.value = value;
View source

**Comment by Alaaeldin-as1405124** [comment]
You're not doing something wrong it's the new version issue. In your package.json use  "zustand": "^3.5.10" instead of the latest version that will solve your issue.

**Comment by dai-shi** [comment]
But, how we can solve the issue with the latest version is a question.

**Comment by dai-shi** [comment]
Does #953 help?

Please try:
```
yarn add https://pkg.csb.dev/pmndrs/zustand/commit/75762b2a/zustand
```

**Comment by Kripu77** [comment]
> Does #953 help?
> 
> Please try:
> 
> ```
> yarn add https://pkg.csb.dev/pmndrs/zustand/commit/75762b2a

> You're not doing something wrong it's the new version issue. In your package.json use "zustand": "^3.5.10" instead of the latest version that will solve your issue.

Hi, I tried this but I still have the same issue.

**Comment by dai-shi** [comment]
Oops, it's not a bug. I now know why.

You should export this:
```js
export const useStore = create(set => ({
  bears: 0,
  increasePopulation: () => set(state => ({ bears: state.bears + 1 })),
  removeAllBears: () => set({ bears: 0 })
}))
```
and import it.

You shouldn't need the following.
```js
import { useStore } from 'zustand'
```

**Comment by Kripu77** [comment]
> Oops, it's not a bug. I now know why.
> 
> 
> 
> You should export this:
> 
> ```js
> 
> export const useStore = create(set => ({
> 
>   bears: 0,
> 
>   increasePopulation: () => set(state => ({ bears: state.bears + 1 })),
> 
>   removeAllBears: () => set({ bears: 0 })
> 
> }))
> 
> ```
> 
> and import it.
> 
> 
> 
> You shouldn't need the following.
> 
> ```js
> 
> import { useStore } from 'zustand'
> 
> ```

Hi Dai, 
Yes I have done export and import of store.

**Comment by dai-shi** [comment]
Can you share a small reproduction code?

**Comment by Kripu77** [comment]
HI Dai-Shi, 
 
My store component:"

**Comment by dai-shi** [comment]
Please do not use `import { useStore } from 'zustand'`. That's for different use cases.

Here you are: https://codesandbox.io/s/suspicious-river-pxq7t5?file=/src/Home.jsx

**Comment by Kripu77** [comment]
Thank you so much for your help. Much appreciated.

**Comment by umararshad4** [comment]
bro simply import your store from the file where your created it not from the zustand!
thats the issue!
Trick: 
      Simply avoid that issue by creating your store with different name not with the useStore.
Enjoy!

**Comment by Kripu77** [comment]
> bro simply import your store from the file where your created it not from the zustand! thats the issue! Trick: Simply avoid that issue by creating your store with different name not with the useStore. Enjoy!

Hey Umar the issue was resolved almost a month ago, I wasn't sure how to use Zustand at that point of time.

**Comment by qwertyMu** [comment]
> bro simply import your store from the file where your created it not from the zustand! thats the issue! Trick: Simply avoid that issue by creating your store with different name not with the useStore. Enjoy!

This is the comment that solved it for me! Thanks!

**Comment by dai-shi** [comment]
As I understand, this is unintentionally bad experience. Changing the export name is not an option. I might try improving docs.

**Comment by dai-shi** [comment]
#1125 
I wonder how much it could help, though.

**Comment by waseem196** [comment]
Unhandled Runtime Error
TypeError: Cannot read properties of null (reading 'getSnapshot')

Call Stack
updateSyncExternalStore
node_modules\react-dom\cjs\react-dom.development.js (16888:0)
Object.useSyncExternalStore
node_modules\react-dom\cjs\react-dom.development.js (17943:0)
useSyncExternalStore
node_modules\react\cjs\react.development.js (1676:0)
eval
node_modules\@preact\signals-react\dist\signals.module.js (1:1033)
Object.set [as current]
node_modules\@preact\signals-react\dist\signals.module.js (1:1084)
renderWithHooks
node_modules\react-dom\cjs\react-dom.development.js (16292:0)
updateForwardRef
node_modules\react-dom\cjs\react-dom.development.js (19226:0)
beginWork
node_modules\react-dom\cjs\react-dom.development.js (21636:0)
beginWork$1
node_modules\react-dom\cjs\react-dom.development.js (27426:0)
performUnitOfWork
node_modules\react-dom\cjs\react-dom.development.js (26557:0)
workLoopSync
node_modules\react-dom\cjs\react-dom.development.js (26466:0)
renderRootSync
node_modules\react-dom\cjs\react-dom.development.js (26434:0)
performSyncWorkOnRoot
node_modules\react-dom\cjs\react-dom.development.js (26085:0)
flushSyncCallbacks
node_modules\react-dom\cjs\react-dom.development.js (12042:0)
eval
node_modules\react-dom\cjs\react-dom.development.js (25651:0)

**Comment by waseem196** [comment]
I am receiving the above error when I @preact/signals npm package

**Comment by dbritto-dev** [comment]
> I am receiving the above error when I use @preact/signals npm package

@waseem196 we can't ensure that something works well with `@preact/signals` since that package is not fully compatible with react -> https://github.com/facebook/react/issues/26704#issuecomment-1522044060

**Comment by waseem196** [comment]
Ok thanks Danilo.

**Comment by quangbeuu** [comment]
You shouldn't import this:

**Comment by quangbeuu** [comment]
You shouldn't import this:

**Comment by quangbeuu** [comment]
You shouldn't import this: `import { useStore } from 'zustand'`, You should import the source file you export useStore, Example: `import { useStore } from "../store";`

**Comment by RshmanGit** [comment]
I think the example in the documentation is a little misleading. 
Because there's a function called useStore in the library and they create a store called use store in the example. People are being led the believe that they're to import useStore instead of their own stores.

https://docs.pmnd.rs/zustand/getting-started/introduction#installation

**Comment by dai-shi** [comment]
Yes, I had the same concern and fixed in #1125 back then.
Would anyone like to send a PR please?
